internal/impl/confluent/serde_hamba_avro_test.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confluent
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+func TestHambaAvroReferences(t *testing.T) {
+	tCtx, done := context.WithTimeout(context.Background(), time.Second*10)
+	defer done()
+
+	rootSchema := `[
+  "benthos.namespace.com.foo",
+  "benthos.namespace.com.bar"
+]`
+
+	fooSchema := `{
+	"namespace": "benthos.namespace.com",
+	"type": "record",
+	"name": "foo",
+	"fields": [
+		{ "name": "Woof", "type": "string"}
+	]
+}`
+
+	barSchema := `{
+	"namespace": "benthos.namespace.com",
+	"type": "record",
+	"name": "bar",
+	"fields": [
+		{ "name": "Moo", "type": "string"}
+	]
+}`
+
+	urlStr := runSchemaRegistryServer(t, func(path string) ([]byte, error) {
+		switch path {
+		case "/subjects/root/versions/latest", "/schemas/ids/1":
+			return mustJBytes(t, map[string]any{
+				"id":         1,
+				"version":    10,
+				"schema":     rootSchema,
+				"schemaType": "AVRO",
+				"references": []any{
+					map[string]any{"name": "benthos.namespace.com.foo", "subject": "foo", "version": 10},
+					map[string]any{"name": "benthos.namespace.com.bar", "subject": "bar", "version": 20},
+				},
+			}), nil
+		case "/subjects/foo/versions/10", "/schemas/ids/2":
+			return mustJBytes(t, map[string]any{
+				"id": 2, "version": 10, "schemaType": "AVRO",
+				"schema": fooSchema,
+			}), nil
+		case "/subjects/bar/versions/20", "/schemas/ids/3":
+			return mustJBytes(t, map[string]any{
+				"id": 3, "version": 20, "schemaType": "AVRO",
+				"schema": barSchema,
+			}), nil
+		}
+		return nil, nil
+	})
+
+	subj, err := service.NewInterpolatedString("root")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name        string
+		input       string
+		output      string
+		errContains []string
+	}{
+		{
+			name:   "a foo",
+			input:  `{"Woof":"hhnnnnnnroooo"}`,
+			output: `{"Woof":"hhnnnnnnroooo"}`,
+		},
+		{
+			name:   "a bar",
+			input:  `{"Moo":"mmuuuuuueew"}`,
+			output: `{"Moo":"mmuuuuuueew"}`,
+		},
```

Comment:
How do these differ?

Reply:
They don't, but this test is just copied over from the goavro tests to ensure we're doing the same thing here